5.5 GA Java Client [PA-512] (#1357)
hazelcast/hz-docs: docs/modules/clients/pages/java.adoc

> client= Java Client

Reviewer: ```suggestion
```

Reviewer 2: removed `client`, retained the `=Java Client` header

> client= Java Client

Reviewer 2: Seems to be a typo. Fixed

Hazelcast provides a {java-client} which you can use to connect to a Hazelcast cluster. `hazelcast-<VERSION>.jar` is bundled in the Hazelcast standard package, so just add `hazelcast-<VERSION>.jar` to your classpath and you can start using this client as if you are using the Hazelcast API.

> If you are interested in using a standalone or lightweight Java client, you can try the {java-client-new}. This client can interact with a Hazelcast cluster without being a full member. Please note that while the {java-client-new} provides full feature parity with the {java-client} in terms of functionality, there are some differences that should be considered before deciding to use it in production environments. For more info, see xref:java#java-client-standalone[].

Reviewer: ```suggestion
If you are interested in using a standalone or lightweight Java client, you can try the {java-client-new}. This client can interact with a Hazelcast cluster without being a full member. For more info on features and differences between the clients, see xref:java#java-client-standalone[].
```

TIP: For a tutorial on getting started with Java in an embedded topology, see xref:getting-started:get-started-java.adoc[].

> === Get started with {java-client-new}

Reviewer: @ihsandemir is this section still correct in terms of dependencies considering v8? Will this section have to expand to cover v8 vs v17?

Reviewer 3: for now we did not change the name of the dependency. we can assume it will be the same and if we change in the future, we can update it.

Reviewer: Thanks Ihsan - I take it we do need to add information on which version of JDK is supported for the standalone client?

To migrate an application from the {java-client} to the {java-client-new}, you only have to update the dependency as described above.

[java-client-standalone]
==== {java-client-new}

The {java-client-new} supports all major features of the {java-client}, but with certain limitations outlined below:

* XA transactions are not supported
* Only built-in sources and sinks are supported by the Pipeline API. External connectors, such as those for cloud storage (e.g., S3, Azure, GCS), databases (e.g., MongoDB, PostgreSQL, MySQL), streaming platforms (e.g., Kafka, Kinesis), and file formats (e.g., Avro, CSV), are not supported  

> * Cloud support includes Hazelcast Cloud, Kubernetes, AWS, Azure, and Google Cloud Platform. Other environments, such as Apache jclouds, Consul, Docker, etcd, Eureka, Heroku, VMware Tanzu (formerly Pivotal Cloud Foundry), and Zookeeper, are not supported

Reviewer: ```suggestion
* Cloud support includes Hazelcast Cloud, Kubernetes, AWS, Azure, and GCP. Other environments, such as Apache jclouds, Consul, Docker, etcd, Eureka, Heroku, VMware Tanzu (formerly Pivotal Cloud Foundry), and Zookeeper, are not supported
```

To migrate an application from the {java-client} to the {java-client-new}, you only have to update the dependency as described above.

[java-client-standalone]
==== {java-client-new}

The {java-client-new} provides full feature parity with the {java-client} in terms of functionality. However, please note the following differences and restrictions:

* Client Configuration Import is not supported.
* XA Transactions are not supported.
* For Pipeline API only built-in sources and sinks are supported.
* Cloud support is only for Hazelcast Cloud and Kubernetes.
* Session replication with Spring Boot is not supported.

> * You cannot use the {java-client} and the {java-client-new} on the same JVM.

Reviewer 2: I have updated the list to reflect the latest changes. Client config import is now supported. Additional details have been added regarding cloud platforms and connectors. The statement about feature parity has been clarified for accuracy.
